### PR TITLE
Added log10 to the list of unary functions df.eval can handle

### PR DESCRIPTION
**Pull request by euri10**
- [x] closes https://github.com/pandas-dev/pandas/issues/24139 
- [x] tests ~~added~~ / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

`(pandas) ➜  pandas git:(binary_math_log) git diff upstream/master -u -- "*.py" | flake8 --diff
`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
can you add a test for this, in pandas/tests/computation/test_eval.py & a whatsnew note (bug fixes for numeric is ok)

**Review by jreback — CHANGES_REQUESTED** [review]
comments

**Comment by euri10** [comment]
Added the whatsnew part as requested @jreback 
For the tests, it seems unecessary to me to add something new, the existing one are already handle the case in [here](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/computation/test_eval.py#L1625-L1633)

File: doc/source/whatsnew/v0.23.5.txt [hunk 1]
```diff
@@ -52,3 +52,7 @@ Bug Fixes
 **I/O**
 
 - Bug in :func:`read_csv` that caused it to raise ``OverflowError`` when trying to use 'inf' as ``na_value`` with integer index column (:issue:`17128`)
+
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.5.txt]: can you move this to 0.24.0

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.23.5.txt (hunk 1 above)
Comment (euri10) [reply doc/source/whatsnew/v0.23.5.txt]: oups, done, sorry for that

File: doc/source/whatsnew/v0.23.5.txt [hunk 2]
```diff
@@ -51,4 +51,4 @@ Bug Fixes
 
 **I/O**
 
-- Bug in :func:`read_csv` that caused it to raise ``OverflowError`` when trying to use 'inf' as ``na_value`` with integer index column (:issue:`17128`)
+- Bug in :func:`read_csv` that caused it to raise ``OverflowError`` when trying to use 'inf' as ``na_value`` with integer index column (:issue:`17128`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.5.txt]: you need to checkout this file from master as it should be unchanged.

File: doc/source/whatsnew/v0.24.0.rst [hunk 3]
```diff
@@ -1375,6 +1375,7 @@ Numeric
 - :meth:`Series.agg` can now handle numpy NaN-aware methods like :func:`numpy.nansum` (:issue:`19629`)
 - Bug in :meth:`Series.rank` and :meth:`DataFrame.rank` when ``pct=True`` and more than 2:sup:`24` rows are present resulted in percentages greater than 1.0 (:issue:`18271`)
 - Calls such as :meth:`DataFrame.round` with a non-unique :meth:`CategoricalIndex` now return expected data. Previously, data would be improperly duplicated (:issue:`21809`).
+- Added log10 to the list of supported functions in ``df.eval`` (:issue:`24139`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.rst]: use double back-ticks on ``log10``, use a ``:meth:`` reference for eval

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green.

**Pull request merged**